### PR TITLE
Unify CLI exit codes and improve err msgs

### PR DESCRIPTION
**Pull request by mike-sul**
- Improve log messages related to updating of the device/local TUF repo.
- Unify exit codes across both CLI utilities, `aklite-offline` and `aktualizr-lite`.
- Add tests to cover changes related to the exit code unification.

### CODE REVIEW

File: src/api.cc [hunk 1]
```diff
@@ -332,33 +328,38 @@ std::vector<TufTarget> toTufTargets(const std::vector<Uptane::Target>& targets)
...
+  } catch (const std::exception& exc) {
+    LOG_ERROR << "Failed to update the local TUF repo: " << exc.what();
+    return CheckInResult{CheckInResult::Status::Failed, "", {}};
+  }
+
+  LOG_INFO << "The local TUF repo has been successfully updated";
+  LOG_INFO << "Searching for matching TUF Targets...";
+  auto matchingTargets = filterTargets(tuf_repo_->GetTargets(), hw_id_, client_->tags, secondary_hwids_);
+  if (matchingTargets.empty()) {
+    LOG_ERROR << "No matching Targets are found for a given device; check a device tag or a hardware ID";
```
Comment (detsch) [inline src/api.cc:344]:
```suggestion
    LOG_ERROR << "There are no Targets matching the current device tag and hardware ID";
```

File: src/api.cc [hunk 2]
```diff
@@ -214,8 +175,43 @@ CheckInResult AkliteClient::CheckIn() const {
...
+  }
+
+  if (check_status == CheckInResult::Status::Failed) {
+    return CheckInResult{check_status, "", {}};
+  }
+
+  LOG_INFO << "Searching for matching TUF Targets...";
+  auto matchingTargets = filterTargets(tuf_repo_->GetTargets(), hw_id_, client_->tags, secondary_hwids_);
+  if (matchingTargets.empty()) {
+    LOG_ERROR << "No matching Targets are found for a given device; check a device tag or a hardware ID";
```
Comment (detsch) [inline src/api.cc:210]:
```suggestion
    LOG_ERROR << "There are no Targets matching the current device tag and hardware ID";
```

**Review by detsch — APPROVED** [review]
LGTM

File: src/api.cc (hunk 2 above)
Comment (mike-sul) [reply src/api.cc:210]: Can we add the verb `found` or synonym to this error message so it goes inline with the `"Searching for matching TUF Targets..." message. 
Maybe something like "Couldn't find..." or "Failed to find"?

Comment (detsch) [reply src/api.cc:210]: Sure, makes sense. Adjusted the suggestions accordingly.

Comment (mike-sul) [reply src/api.cc]: updated via force push

**Pull request merged**